<?xml version="1.0"?>
<launch>
  <!-- Arguments -->
  <arg name="robot_model" />
  <arg name="max_robot_speed" />
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="/zed/zed_node/odom" />
  <arg name="move_forward_only" default="false"/>


  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <!-- costmaps -->-->
    <rosparam file="$(find zedbot_people_tracking)/params/move_base_params/costmap_common_params_$(arg robot_model).yaml" command="load" ns="global_costmap" /> 
    <rosparam file="$(find zedbot_people_tracking)/params/move_base_params/costmap_common_params_$(arg robot_model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find zedbot_people_tracking)/params/move_base_params/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find zedbot_people_tracking)/params/move_base_params/global_costmap_params.yaml" command="load" />

    <!-- planners -->-->
    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <param name="base_global_planner" value="global_planner/GlobalPlanner" />
    <param name="TebLocalPlannerROS/max_vel_x" value="$(arg max_robot_speed)" />
    <param name="TebLocalPlannerROS/acc_lim_x" value="0.02" />

    <rosparam file="$(find zedbot_people_tracking)/params/move_base_params/move_base_params.yaml" command="load" />
    <rosparam file="$(find zedbot_people_tracking)/params/move_base_params/global_planner_params_$(arg robot_model).yaml" command="load" />
    <rosparam file="$(find zedbot_people_tracking)/params/move_base_params/teb_local_planner_params_$(arg robot_model).yaml" command="load" />
    
    <!-- Remap -->-->
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
  </node>
</launch>

